nativelib/src/main/scala/scala/scalanative/runtime/dwarf/BinaryFile.scala

```diff
@@ -0,0 +1,84 @@
+package scala.scalanative.runtime.dwarf
+
+import java.io.DataInputStream
+import Endianness.LITTLE
+import Endianness.BIG
+import java.io.RandomAccessFile
+import java.nio.channels.Channels
+import scalanative.unsigned._
+import MachO._
+import scala.collection.mutable.ArrayBuffer
+import java.io.File
+import java.io.BufferedInputStream
+import java.io.InputStream
+
+class SeekableBufferedInputStream(
```

Comment:
Custom BufferedInputStream that has an capability of seeking the position manually.
I developed this since `RandomAccessFile` is extremely slow due to it's lack of buffer.